File: frontend/src/app/shared/services/operation.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {HttpHelper} from './http-helper';

@Injectable({
  providedIn: 'root'
})
export class OperationService {

  constructor(private http: HttpClient) { }

  createOperation(operationData: any) {
    return this.http.post(`${HttpHelper.baseURL}/operation/create`, operationData);
  }

  getOperations(month?: number, year?: number) {
    if (month && year) {
      const params = new HttpParams().set('month', (month + 1).toString()).set('year', year.toString());
      return this.http.get(`${HttpHelper.baseURL}/operation/get`, {params});
    } else {
      return this.http.get(`${HttpHelper.baseURL}/operation/get`);
    }
  }

  deleteOperation(operationId: number) {
    return this.http.delete(`${HttpHelper.baseURL}/operation/delete/${operationId}`);
  }

  createBudgetPlan(budget: any) {
    return this.http.post(`${HttpHelper.baseURL}/budget/create`, budget);
  }
}

File: backend/src/main/kotlin/com/pnowak/personalbudget/backend/service/interfaces/OperationService.kt
package com.pnowak.personalbudget.backend.service.interfaces

import com.pnowak.personalbudget.backend.dto.CreateOperationDTO
import com.pnowak.personalbudget.backend.entities.Operation

interface OperationService {
    fun createOperation(createOperationDTO: CreateOperationDTO, userId: Long?): Long
    fun getOperations(month: Int?, year: Int?, userIdFromContext: Long): List<Operation>
    fun getOperations(userIdFromContext: Long): List<Operation>
    fun deleteOperation(operationId: Long)
}
